adds Htslib Record to StrandedBed3 conversion

sathish-t opened the pull request:
* This should enable usage of BAM in place of BED in downstream applications
* Added the needed rust-htslib and bio-types packages to Cargo.toml

sathish-t committed:
adds Htslib Record to StrandedBed3 conversion

* This should enable usage of BAM in place of BED in downstream
applications
* Added the needed rust-htslib and bio-types packages to Cargo.toml

sathish-t commented:
I looked into implementing Coordinates directly for the Rust Htslib Record, but it doesn't quite work due to the update_start and update_end functions which will make the BAM data invalid.
If you are happy with this commit, I can add a BAM file somewhere in the repo and add code + documentation on how to intersect it with a bed file by converting it to a StrandedBed3. I will need some guidance on where to put these things.

noamteyssier committed:
dep: gate htslib behind feature flag

noamteyssier committed:
refactor: check strandedness from htslib builtin and avoid extra dep

noamteyssier commented:
hey @sathish-t 

Looks great! Appreciate the addition. 

I made some small edits to this, mainly to gate `htslib` behind a feature flag and to use the builtin reverse check to avoid the extra dependency. 

If you wanted to add code / documentation perhaps we can start an `examples/` directory and you can put your code in there?

sathish-t committed:
adds sample bam file, example code to doc, testing

sathish-t commented:
Hi @noamteyssier,

I've made an examples directory with a small bam file, added a doctest of loading the bam file, a test of loading and intersect with the bam file, and modified the READMEs (README.md and src/lib.rs). The tests I've added are all in bedrs/src/types/record/stranded_genomic_interval.rs. Please let me know what you think!

noamteyssier committed:
refactor(htslib): remove htslib from default features and gate all imports

noamteyssier committed:
ci: added htslib testing to ci

noamteyssier commented:
thanks for the P.R @sathish-t !

Will merge this into main in two steps. Will first move temporary branch and clean up some linting and CI, then will merge into main. 

Cheers!
